sycl/include/CL/sycl/detail/program_impl.hpp

```diff
@@ -93,8 +96,11 @@ class program_impl {
     }
   }
 
+  // Assume OpenCL program is built from C OpenCL source and thus
```

Comment:
```suggestion
  // Assume OpenCL program is built from OpenCL C source and thus
```
The language is called "OpenCL C", but I'm not sure it's a good idea to refer to OpenCL at this level.
Probably it's better to say that programs created by interoperability constructions do not cache kernels.

BTW, are we going to relax this restriction in the future?

Reply:
> BTW, are we going to relax this restriction in the future?
I'm not sure. It's not easy to do and there are no use cases.

Reply:
@s-kanaev, could you add a description to the PR with clarification of limitations, please?
I've seen something in the code comments, but it's useful to have in the log message too. I'm not sure I clearly understand in what cases kernel caching is enabled and when it's not.